front/src/utils/notification.util.ts

```diff
@@ -371,41 +371,47 @@ export const updateArticlesNotification = async (): Promise<void> => {
-      )) as string[] | undefined) ?? null;
-    const currentStep = (await StorageUtils.getObjectValue(
-      StorageKeysConstants.currentStep
-    )) as Step | null;
-
-    if (content) {
-      await cancelAllNotificationsByType(NotificationType.articles);
-      const hasBeenAlreadyNotified =
-        stepsAlreadyCongratulatedForArticles?.includes(
-          currentStep ? currentStep.id.toString() : ""
-        );
-      if (!hasBeenAlreadyNotified) {
-        await sendNotificationReminder(content, trigger);
-
-        // Enregistre les étapes pour lesquelles la notification de félicitations (articles tous lus) a déjà été programmée
-        if (nbArticlesToRead === 0 && currentStep) {
-          const currentStepId = currentStep.id.toString();
-          const newValue = stepsAlreadyCongratulatedForArticles
-            ? stepsAlreadyCongratulatedForArticles.push(currentStepId)
-            : [currentStepId];
-          await StorageUtils.storeObjectValue(
-            StorageKeysConstants.stepsAlreadyCongratulatedForArticles,
-            newValue
+  const isToggleActive = (await StorageUtils.getObjectValue(
+    StorageKeysConstants.notifToggleArticles
+  )) as boolean;
+
+  if (isToggleActive) {
+    const nbArticlesToRead: number = await countCurrentStepArticlesNotRead();
+    if (nbArticlesToRead >= 0) {
+      const trigger: NotificationTriggerInput =
+        nbArticlesToRead > 0
+          ? notifTrigger ?? (await getNewTriggerForArticlesNotification())
+          : NotificationConstants.MIN_TRIGGER;
+      const content = await buildArticlesNotificationContent(nbArticlesToRead);
+
+      const stepsAlreadyCongratulatedForArticles =
+        ((await StorageUtils.getObjectValue(
+          StorageKeysConstants.stepsAlreadyCongratulatedForArticles
+        )) as string[] | undefined) ?? null;
+      const currentStep = (await StorageUtils.getObjectValue(
+        StorageKeysConstants.currentStep
+      )) as Step | null;
+
+      if (content) {
+        await cancelAllNotificationsByType(NotificationType.articles);
+        const hasBeenAlreadyNotified =
+          stepsAlreadyCongratulatedForArticles?.includes(
+            currentStep ? currentStep.id.toString() : ""
           );
+        if (!hasBeenAlreadyNotified) {
+          await sendNotificationReminder(content, trigger);
+
+          // Enregistre les étapes pour lesquelles la notification de félicitations (articles tous lus) a déjà été programmée
+          if (nbArticlesToRead === 0 && currentStep) {
+            const currentStepId = currentStep.id.toString();
+            const newValue = stepsAlreadyCongratulatedForArticles
+              ? stepsAlreadyCongratulatedForArticles.push(currentStepId)
+              : [currentStepId];
+            await StorageUtils.storeObjectValue(
+              StorageKeysConstants.stepsAlreadyCongratulatedForArticles,
+              newValue
+            );
+          }
```

Comment:
idem ici, je pense que ça pourrait être sorti parce qu'il y a beaucoup de choses dans la méthode parente

Reply:
et du coup my guess is : c'est pas testé ?